### PR TITLE
Booking edit fixes

### LINKED ISSUE #3856: Fix toggle in booking edit dialog
**To Reproduce**
1. Go to booking details dialog
2. Edit the booking so the conflict message shows up
3. Open the date picker
4. Toggle element shows up on top of the calendar.

**Expected behavior**
Toggle element should be hidden.

**Screenshots**
image

### LINKED ISSUE #3859: Booking edit: fix calendars when booking is prolonged
**Example:**

Original booking:
image

After end date changed from 02/04 to 04/04 and end time changed from 14:15 to 15:00
image

Two problems occur:
1) original booking calendar should end on 02/04, but it ends on 04/04
2) new booking doesn't display any other bookings, blocking, unbookable periods beyond the original date range (03/04, 04/04)

### PR DESCRIPTION
**Pull request by mic4ael**
fixes #3859 
fixes #3856

### CODE REVIEW

**Comment by mic4ael** [comment]
**Still need to make sure that everything works as expected!**

File: indico/modules/rb_new/client/js/common/bookings/BookingEdit.jsx [hunk 1]
```diff
@@ -200,10 +206,13 @@ class BookingEdit extends React.Component {
...
                     data: {
                         ...currentBooking.data,
+                        pendingCancellations: {},
                         candidates,
                         conflicts,
                         conflictingCandidates,
-                        pendingCancellations: {},
+                        blockings,
+                        unbookableHours,
+                        unbookablePeriods,
```
Comment (meluru) [inline indico/modules/rb_new/client/js/common/bookings/BookingEdit.jsx]: Other bookings are still missing here.

Comment (mic4ael) [reply indico/modules/rb_new/client/js/common/bookings/BookingEdit.jsx]: Yeha, I know. I am still thinking how to properly fix.

Comment (meluru) [reply indico/modules/rb_new/client/js/common/bookings/BookingEdit.jsx]: Ah, sure, I thought this was ready.

Comment (mic4ael) [reply indico/modules/rb_new/client/js/common/bookings/BookingEdit.jsx]: No problem. I should have created a Draft PR ;)

Comment (mic4ael) [reply indico/modules/rb_new/client/js/common/bookings/BookingEdit.jsx]: But unfortunately, once you create a PR you can't make convert it to a draft.

Comment (meluru) [reply indico/modules/rb_new/client/js/common/bookings/BookingEdit.jsx]: Yeah, it's a pity. You could always list WIP items in PR description

**Pull request closed without merging**